nym-authenticator-client/src/lib.rs

```diff
@@ -158,10 +159,14 @@ impl AuthenticatorClient {
         &mut self,
         controller: &dyn BandwidthTicketProvider,
         ticketbook_type: TicketType,
-    ) -> Result<GatewayData> {
+    ) -> std::result::Result<GatewayData, Error> {
         debug!("Registering with the wg gateway...");
         let init_message = match self.auth_version {
```

Comment:
would it perhaps make sense to split this function into 2: one for all things happening before registration (i.e. `Error::NoCredentialSpent` and after (i.e. `Error::CredentialSpent`)?

then we could do something like
```
let foo = stuff_before(...).await.map_err(Error::NoCredentialSpent)?;
let bar = stuff_after(foo).await.map_err(Error::CredentialSpent)?;
...
```